File: Accounting/Accounting.Web/App/services/models/transactionSearch/TransactionSearchRequest.js
/******************************************************************************
* Copyright (C) 2010-2016 GlobalTranz - All Rights Reserved. 
*
* Proprietary and confidential. Unauthorized copying of this file, via any
* medium is strictly prohibited without the explicit permission of GlobalTranz. 
******************************************************************************/

/// <reference path="../../../../Scripts/TypeDefs/durandal.d.ts" />
/// <reference path="../TypeDefs/TransactionSearchModel.d.ts" />
define(["require", "exports", 'durandal/system'], function(require, exports, __refSystem__) {
    var refSystem = __refSystem__;

    /* File Created: Sep 12, 2014
    ** Created By: Satish
    */
    (function (Model) {
        var TransactionSearchRequest = (function () {
            function TransactionSearchRequest(args) {
                this.TransactionSearchType = refSystem.isObject(args) ? args.TransactionSearchType : -1;
                this.Amount = refSystem.isObject(args) ? args.Amount : -1;
                this.OrderStatus = refSystem.isObject(args) ? args.OrderStatus : -1;
                this.BolNumber = refSystem.isObject(args) ? args.BolNumber : "";
                this.ShipperCompanyName = refSystem.isObject(args) ? args.ShipperCompanyName : "";
                this.ConsigneeCompanyName = refSystem.isObject(args) ? args.ConsigneeCompanyName : "";
                this.VendorName = refSystem.isObject(args) ? args.VendorName : "";
                this.CarrierServiceMode = refSystem.isObject(args) ? args.CarrierServiceMode : -1;
                this.NewShipmentType = refSystem.isObject(args) ? args.NewShipmentType : -1;
                this.PoNumber = refSystem.isObject(args) ? args.PoNumber : "";
                this.ShipperCity = refSystem.isObject(args) ? args.ShipperCity : "";
                this.ConsigneeCity = refSystem.isObject(args) ? args.ConsigneeCity : "";
                this.SalesRepName = refSystem.isObject(args) ? args.SalesRepName : "";
                this.SalesRepId = refSystem.isObject(args) ? args.SalesRepId : -1;
                this.ShipperZipCode = refSystem.isObject(args) ? args.ShipperZipCode : "";
                this.ConsigneeZipCode = refSystem.isObject(args) ? args.ConsigneeZipCode : "";
                this.CustomerBolNumber = refSystem.isObject(args) ? args.CustomerBolNumber : "";
                this.ProNumber = refSystem.isObject(args) ? args.ProNumber : "";
                this.ItemNumbers = refSystem.isObject(args) ? args.ItemNumbers : "";
                this.CustomerName = refSystem.isObject(args) ? args.CustomerName : "";
                this.TruckloadQuoteNumber = refSystem.isObject(args) ? args.TruckloadQuoteNumber : '';
                this.InvoiceStatus = refSystem.isObject(args) ? args.InvoiceStatus : -1;
                this.ReferenceNumber = refSystem.isObject(args) ? args.ReferenceNumber : "";
                this.TotalWeight = refSystem.isObject(args) ? args.TotalWeight : -1;
                this.TotalPiece = refSystem.isObject(args) ? args.TotalPiece : -1;
                this.DateType = refSystem.isObject(args) ? args.DateType : -1;
                this.FromDate = refSystem.isObject(args) ? args.FromDate : new Date();
                this.ToDate = refSystem.isObject(args) ? args.ToDate : new Date();
                this.PageSize = refSystem.isObject(args) ? args.PageSize : 10;
                this.PageNumber = refSystem.isObject(args) ? args.PageNumber : 1;
                this.SortCol = refSystem.isObject(args) ? args.SortCol : "";
                this.SortOrder = refSystem.isObject(args) ? args.SortOrder : "ASC";
                this.IsPurchaseOrder = refSystem.isObject(args) ? args.IsPurchaseOrder : 2;
                this.CompanyName = refSystem.isObject(args) ? args.CompanyName : "";
            }
            return TransactionSearchRequest;
        })();
        Model.TransactionSearchRequest = TransactionSearchRequest;
    })(exports.Model || (exports.Model = {}));
    var Model = exports.Model;
});

File: Accounting/Accounting.Web/Models/TransactionSearchRequest.cs
/******************************************************************************
* Copyright (C) 2010-2016 GlobalTranz - All Rights Reserved. 
*
* Proprietary and confidential. Unauthorized copying of this file, via any
* medium is strictly prohibited without the explicit permission of GlobalTranz. 
******************************************************************************/

using System;

namespace Accounting.Web.Models
{
	/// <summary>
	/// Transaction Search Parameters for Transaction Search
	/// </summary>
	/// <createDetails>
	/// <id>US21085</id> <by>Janakiram</by> <date>31-03-2016</date> <description>Download Transaction SearchSettings.</description>		
	/// </createDetails>
	public class TransactionSearchRequest
	{
		/// <summary>
		/// Gets or sets TransactionSearchType
		/// </summary>
		public int TransactionSearchType { get; set; }

		/// <summary>
		/// Gets or sets Amount
		/// </summary>
		public decimal Amount { get; set; }

		/// <summary>
		/// Gets or sets Order Status
		/// </summary>
		public int OrderStatus { get; set; }

		/// <summary>
		/// Gets or sets BolNumber
		/// </summary>
		public string BolNumber { get; set; }

		/// <summary>
		/// Gets or sets ShipperCompanyName
		/// </summary>
		public string ShipperCompanyName { get; set; }

		/// <summary>
		/// Gets or sets ConsigneeCompanyName
		/// </summary>
		public string ConsigneeCompanyName { get; set; }

		/// <summary>
		/// Gets or sets VendorName
		/// </summary>
		public string VendorName { get; set; }

		/// <summary>
		/// Gets or sets CarrierServiceMode
		/// </summary>
		public int CarrierServiceMode { get; set; }

		/// <summary>
		/// Gets or sets NewShipmentType
		/// </summary>
		public int NewShipmentType { get; set; }

		/// <summary>
		/// Gets or sets PoNumber
		/// </summary>
		public string PoNumber { get; set; }

		/// <summary>
		/// Gets or sets ShipperCity
		/// </summary>
		public string ShipperCity { get; set; }

		/// <summary>
		/// Gets or sets ConsigneeCity
		/// </summary>
		public string ConsigneeCity { get; set; }

		/// <summary>
		/// Gets or sets SalesRepName
		/// </summary>
		public string SalesRepName { get; set; }

		/// <summary>
		/// Gets or sets SalesRepId
		/// </summary>
		public int SalesRepId { get; set; }

		/// <summary>
		/// Gets or sets ShipperZipCode
		/// </summary>
		public string ShipperZipCode { get; set; }

		/// <summary>
		/// Gets or sets ConsigneeZipCode
		/// </summary>
		public string ConsigneeZipCode { get; set; }

		/// <summary>
		/// Gets or sets CustomerBolNumber
		/// </summary>
		public string CustomerBolNumber { get; set; }

		/// <summary>
		/// Gets or sets ProNumber
		/// </summary>
		public string ProNumber { get; set; }

		/// <summary>
		/// Gets or sets ItemNumbers
		/// </summary>
		public string ItemNumbers { get; set; }

		/// <summary>
		/// Gets or sets CustomerName
		/// </summary>
		public string CustomerName { get; set; }

		/// <summary>
		/// Gets or sets TruckloadQuoteNumber
		/// </summary>
		public string TruckloadQuoteNumber { get; set; }

		/// <summary>
		/// Gets or sets InvoiceStatus
		/// </summary>
		public int InvoiceStatus { get; set; }

		/// <summary>
		/// Gets or sets ReferenceNumber
		/// </summary>
		public string ReferenceNumber { get; set; }

		/// <summary>
		/// Gets or sets TotalWeight
		/// </summary>
		public int TotalWeight { get; set; }

		/// <summary>
		/// Gets or sets TotalPiece
		/// </summary>
		public int TotalPiece { get; set; }

		/// <summary>
		/// Gets or sets DateType
		/// </summary>
		public int DateType { get; set; }

		/// <summary>
		/// Gets or sets FromDate
		/// </summary>
		public DateTime FromDate { get; set; }

		/// <summary>
		/// Gets or sets ToDate
		/// </summary>
		public DateTime ToDate { get; set; }

		/// <summary>
		/// Gets or sets PageSize
		/// </summary>
		public int PageSize { get; set; }

		/// <summary>
		/// Gets or sets PageNumber
		/// </summary>
		public int PageNumber { get; set; }

		/// <summary>
		/// Gets or sets the values for SortOrder
		/// </summary>
		public string SortOrder { get; set; }

		/// <summary>
		/// Gets or sets the values for SortCol
		/// </summary>
		public string SortCol { get; set; }

		/// <summary>
		/// Gets or Sets Purchase order flag
		/// </summary>
		public SearchTransactionType IsPurchaseOrder { get; set; }

		/// <summary>
		/// Gets or sets the values for CompanyName
		/// </summary>
		public string CompanyName { get; set; }
		
		/// <summary>
		/// Search Transaction Type
		/// </summary>
		public enum SearchTransactionType
		{
			/// <summary>
			/// PurchaseOrders
			/// </summary>
			PurchaseOrders,

			/// <summary>
			/// VendorBills
			/// </summary>
			VendorBills,

			/// <summary>
			/// Both
			/// </summary>
			Both
		}
	}
}